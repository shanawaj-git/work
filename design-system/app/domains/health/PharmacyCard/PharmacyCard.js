import React from 'react';
import PropTypes from 'prop-types';
import { v4 as uuidV4 } from 'uuid';

import CardBody from 'molecules/Card/CardBody';
import {
  StyledCardHeader,
  StyledAccordion,
  StyledAccordionSummary,
  StyledAccordionDetails,
  StyledCardFooter,
  StyledDivider,
  PriceSummary,
  Container,
} from './styledComponents';

export const PharmacyCard = props => {
  const { bodyDetails, bodySummary, footerDetails, expanded, disabled } = props;
  const [isExpanded, setExpanded] = React.useState(expanded);

  const toggleSeperator = () => {
    setExpanded(!isExpanded);
  };

  return (
    <>
      <StyledAccordion
        onChange={toggleSeperator}
        expanded={isExpanded}
        disabled={disabled}
      >
        <StyledAccordionSummary>
          <StyledCardHeader {...props} />
        </StyledAccordionSummary>
        <StyledAccordionDetails>
          <CardBody details={bodyDetails} />
        </StyledAccordionDetails>
      </StyledAccordion>
      <StyledDivider />
      <Container>
        {isExpanded &&
          bodySummary &&
          bodySummary.map(footerDetail => {
            const { text, sideText } = footerDetail;
            return (
              <PriceSummary text={text} key={uuidV4()} sideText={sideText} />
            );
          })}
      </Container>

      <StyledCardFooter details={footerDetails} shouldShowButton={isExpanded} />
    </>
  );
};

PharmacyCard.propTypes = {
  bodyDetails: PropTypes.arrayOf(
    PropTypes.shape({
      icon: PropTypes.string,
      text: PropTypes.string,
      sideText: PropTypes.string,
    }),
  ),
  bodySummary: PropTypes.arrayOf(
    PropTypes.shape({
      icon: PropTypes.string,
      text: PropTypes.string,
      sideText: PropTypes.string,
    }),
  ),
  footerDetails: PropTypes.shape({
    buttonText: PropTypes.string,
    titleText: PropTypes.string,
    titleSideText: PropTypes.string,
    caption: PropTypes.string,
  }),
  expanded: PropTypes.bool,
  disabled: PropTypes.bool,
};
