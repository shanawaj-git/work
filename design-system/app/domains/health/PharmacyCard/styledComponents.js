import styled from 'styled-components';
import React from 'react';
import {
  Accordion,
  AccordionDetails,
  AccordionSummary,
  Divider,
} from '@mui/material';
import { styled as MuiStyled } from '@mui/material/styles';
import CardFooter from 'molecules/Card/CardFooter';
import CardHeader from 'molecules/Card/CardHeader';
import { colorSelector } from 'themes';

import { TextWithIcon } from 'molecules/TextWithIcon/TextWithIcon';

export const StyledCardHeader = styled(props => <CardHeader {...props} />)(
  () => ({
    '.card-header-wrapper': {
      display: 'flex',
      flexDirection: 'column',

      '> div': {
        display: 'flex',
      },

      '.card-header-footer': {
        marginTop: '20px',
        display: 'block',
        '.MuiTypography-caption': {
          fontSize: '16px',
          color: colorSelector('c100'),
          fontWeight: 600,
        },
      },
    },

    '.card-header-image': {
      display: 'flex',
      width: '40px',
      height: '40px',
      borderRadius: '8px',
      justifyContent: 'space-between',
      marginRight: '8px',
    },

    '.textWithIcon': {
      display: 'flex',

      img: {
        marginRight: '4px',
        height: '16px',
        width: '16px',
        marginTop: 'auto',
        marginBottom: 'auto',
      },
    },
  }),
);

export const StyledAccordion = MuiStyled(Accordion)`
&.MuiAccordion-root {
	box-shadow: none;
	margin-bottom: 0px;
	:before {
		opacity: 0;
	}
}

  .MuiAccordion-root:before {
    content: '' !important;
    opacity: 0;
  }
`;

export const StyledAccordionDetails = styled(props => (
  <AccordionDetails {...props} />
))(() => ({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'flex-start',
  alignItems: 'normal',

  '&&.MuiAccordionDetails-root': {
    paddingBottom: '0px',
  },

  '.textWithIcon': {
    display: 'flex',
    justifyContent: 'center',
    height: '35px',
    img: {
      height: '16px',
      width: '16px',
      marginRight: '8px',
    },

    '.MuiTypography-caption': {
      color: colorSelector('primaryFont'),
      fontWeight: '400',
      fontSize: '13px',
    },

    '.textWithIcon-sideText': {
      marginLeft: 'auto',
    },
  },
}));

export const StyledAccordionSummary = MuiStyled(AccordionSummary)`
	.MuiAccordionSummary-content {
		display: flex;
		flex-direction: column;

		&.Mui-expanded {
			margin-bottom: 10px;
		}
	}
	.textWithIcon {
		img {
			margin-right: 8px;
		}
	}
  }
`;

export const StyledCardFooter = styled(props => <CardFooter {...props} />)(
  () => ({
    paddingLeft: '15px',
    paddingRight: '15px',
    display: 'flex',
    flexDirection: 'column',

    '.card-footer-title': {
      marginBottom: '10px',

      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-between',

      '.MuiTypography-caption': {
        fontSize: '15px',
        fontWeight: '600',
      },

      'span:last-child': {
        color: colorSelector('c100'),
        textAlignment: 'center',
        background: colorSelector('p100'),
        borderRadius: '20px',
        padding: '10px',
      },
    },
    '.card-footer-subtitle': {
      paddingBottom: '20px',
      color: colorSelector('c100'),
    },
  }),
);

export const StyledDivider = MuiStyled(Divider)`
  margin-bottom: 10px;
  width: 97%;
  margin-left: auto;
  margin-right: auto;
`;

export const PriceSummary = styled(TextWithIcon)`
  font-size: 13px;
  font-weight: 400;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 35px;
`;

export const Container = styled.div`
  padding-left: 15px;
  padding-right: 15px;
  margin-bottom: 16px;
`;
