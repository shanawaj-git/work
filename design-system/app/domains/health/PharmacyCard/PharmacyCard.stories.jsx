/* eslint-disable react/prop-types */

import React from 'react';
import { PharmacyCard } from './PharmacyCard';

import StoryWrapper from 'molecules/utility/StoryWrapper';

export default {
  title: 'Components/Containers/cards/PharmacyCard',
  component: PharmacyCard,
  decorators: [Story => <StoryWrapper>{Story()}</StoryWrapper>],
};

const tickIcon = require('images/tick.svg');

const defaultProps = {
  title: 'Al Manara Pharmacy',
  subTitle: 'Al Sufouh',
  description: 'Open until 22:00',
  subDescription: '20 min',
  subDescriptionIcon: require('images/truck.svg'),
  headerImage: require('images/almanara-pharmacy.png'),
  footerIcon: tickIcon,
  footerText: 'All meds covered',
  bodyDetails: [
    {
      text: 'Hyrdoxyzine hydrochloride',
      sideText: 'AED 150.00',
      icon: tickIcon,
    },
    {
      text: 'Glimepiride-rosiglitazone',
      sideText: 'AED 150.00',
      icon: tickIcon,
    },
    { text: 'Novolin 70/30', sideText: 'AED 150.00', icon: tickIcon },
  ],
  bodySummary: [
    {
      text: 'Cost of Medicines',
      sideText: 'AED 550.00',
    },
    {
      text: 'Est. Insurance Coverage*',
      sideText: 'AED 500.00',
    },
  ],
  footerDetails: {
    buttonText: 'Select pharmacy',
    titleText: 'Your est. co-payment*',
    titleSideText: 'AED 50.00',
    caption: '*Final amounts subject to insurance coverage',
  },
};

const render = props => (
  <>
    <PharmacyCard {...props} />
  </>
);

const Template = args => (
  <StoryWrapper themeVariant={args.themeType}>{render(args)}</StoryWrapper>
);

export const expanded = Template.bind({});
expanded.args = {
  ...defaultProps,
  expanded: true,
};

export const collapsed = Template.bind({});
collapsed.args = {
  ...defaultProps,
};

export const disabled = Template.bind({});
disabled.args = {
  ...defaultProps,
  disabled: true,
};
