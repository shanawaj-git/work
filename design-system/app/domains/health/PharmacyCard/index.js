import { PharmacyCard } from './PharmacyCard';

export default PharmacyCard;
