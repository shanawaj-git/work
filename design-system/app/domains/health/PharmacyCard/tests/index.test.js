import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { render, unmountComponentAtNode } from 'react-dom';
import renderer, { act } from 'react-test-renderer';

import PharmacyCard from '../index';

const shallowRenderer = new ShallowRenderer();
let container = null;

describe('Input.js', () => {
  const tickIcon = require('../../../../images/tick.svg');
  const props = {
    title: 'Al Manara Pharmacy',
    subTitle: 'Al Sufouh',
    description: 'Open until 22:00',
    subDescription: '20 min',
    subDescriptionIcon: require('../../../../images/truck.svg'),
    headerImage: require('../../../../images/almanara-pharmacy.png'),
    footerIcon: tickIcon,
    footerText: 'All meds covered',
    bodyDetails: [
      {
        text: 'Hyrdoxyzine hydrochloride',
        sideText: 'AED 150.00',
        icon: tickIcon,
      },
      {
        text: 'Glimepiride-rosiglitazone',
        sideText: 'AED 150.00',
        icon: tickIcon,
      },
      { text: 'Novolin 70/30', sideText: 'AED 150.00', icon: tickIcon },
    ],
    bodySummary: [
      {
        text: 'Cost of Medicines',
        sideText: 'AED 550.00',
      },
      {
        text: 'Est. Insurance Coverage*',
        sideText: 'AED 500.00',
      },
    ],
    footerDetails: {
      buttonText: 'Select pharmacy',
      titleText: 'Your est. co-payment*',
      titleSideText: 'AED 50.00',
      caption: '*Final amounts subject to insurance coverage',
    },
  };

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  describe('index', () => {
    it('Should import succesfully', () => {
      const tree = renderer.create(<PharmacyCard {...props} />);
      expect(tree).toBeTruthy();
      expect(tree).toMatchSnapshot();
    });

    it('Should display a disabled card', () => {
      shallowRenderer.render(<PharmacyCard {...props} />);
      const renderedOutput = shallowRenderer.getRenderOutput();

      expect(renderedOutput).toMatchSnapshot();
    });

    it('should open/close accordion when clicking on the accordion summary', () => {
      act(() => {
        render(<PharmacyCard {...props} />, container);
      });

      // get a hold of the input element, and trigger some clicks on it
      const accordion = document.querySelector('.MuiAccordionSummary-root');

      act(() => {
        accordion.dispatchEvent(new MouseEvent('click', { bubbles: true }));
      });

      expect(accordion.getAttribute('aria-expanded')).toEqual('true');

      act(() => {
        accordion.dispatchEvent(new MouseEvent('click', { bubbles: true }));
      });

      expect(accordion.getAttribute('aria-expanded')).toEqual('false');
    });
  });
});
