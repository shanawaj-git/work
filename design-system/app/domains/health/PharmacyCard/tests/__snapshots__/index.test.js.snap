// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.js index Should display a disabled card 1`] = `
<React.Fragment>
  <ForwardRef(Styled(Component))
    onChange={[Function]}
  >
    <ForwardRef(Styled(Component))>
      <ForwardRef(Styled(Component))
        bodyDetails={
          Array [
            Object {
              "icon": "IMAGE_MOCK",
              "sideText": "AED 150.00",
              "text": "Hyrdoxyzine hydrochloride",
            },
            Object {
              "icon": "IMAGE_MOCK",
              "sideText": "AED 150.00",
              "text": "Glimepiride-rosiglitazone",
            },
            Object {
              "icon": "IMAGE_MOCK",
              "sideText": "AED 150.00",
              "text": "Novolin 70/30",
            },
          ]
        }
        bodySummary={
          Array [
            Object {
              "sideText": "AED 550.00",
              "text": "Cost of Medicines",
            },
            Object {
              "sideText": "AED 500.00",
              "text": "Est. Insurance Coverage*",
            },
          ]
        }
        description="Open until 22:00"
        footerDetails={
          Object {
            "buttonText": "Select pharmacy",
            "caption": "*Final amounts subject to insurance coverage",
            "titleSideText": "AED 50.00",
            "titleText": "Your est. co-payment*",
          }
        }
        footerIcon="IMAGE_MOCK"
        footerText="All meds covered"
        headerImage="IMAGE_MOCK"
        subDescription="20 min"
        subDescriptionIcon="IMAGE_MOCK"
        subTitle="Al Sufouh"
        title="Al Manara Pharmacy"
      />
    </ForwardRef(Styled(Component))>
    <ForwardRef(Styled(Component))>
      <CardBody
        details={
          Array [
            Object {
              "icon": "IMAGE_MOCK",
              "sideText": "AED 150.00",
              "text": "Hyrdoxyzine hydrochloride",
            },
            Object {
              "icon": "IMAGE_MOCK",
              "sideText": "AED 150.00",
              "text": "Glimepiride-rosiglitazone",
            },
            Object {
              "icon": "IMAGE_MOCK",
              "sideText": "AED 150.00",
              "text": "Novolin 70/30",
            },
          ]
        }
      />
    </ForwardRef(Styled(Component))>
  </ForwardRef(Styled(Component))>
  <ForwardRef(Styled(Component)) />
  <ForwardRef(styledComponents__Container) />
  <ForwardRef(Styled(Component))
    details={
      Object {
        "buttonText": "Select pharmacy",
        "caption": "*Final amounts subject to insurance coverage",
        "titleSideText": "AED 50.00",
        "titleText": "Your est. co-payment*",
      }
    }
  />
</React.Fragment>
`;

exports[`Input.js index Should import succesfully 1`] = `
Array [
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1yr3319-MuiPaper-root-MuiAccordion-root"
  >
    <div
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-2k2ln8-MuiButtonBase-root-MuiAccordionSummary-root"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
      >
        <div
          className="sc-bdVaJa jJSDIg"
        >
          <div
            className="card-header-wrapper"
          >
            <div>
              <div
                className="card-header-image"
              >
                <img
                  alt="Al Manara Pharmacy"
                  src="IMAGE_MOCK"
                />
              </div>
              <div
                className="card-header-body"
              >
                <h5
                  className="MuiTypography-root MuiTypography-h5 card-header-title css-11estw0-MuiTypography-root"
                >
                  Al Manara Pharmacy
                </h5>
                <div
                  className="textWithIcon card-header-subTitle"
                >
                  <span
                    className="MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root"
                  >
                    Al Sufouh
                  </span>
                </div>
                <div
                  className="textWithIcon card-header-description"
                >
                  <span
                    className="MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root"
                  >
                    Open until 22:00
                  </span>
                </div>
                <div
                  className="textWithIcon card-header-subDescription"
                >
                  <img
                    alt="20 min"
                    src="IMAGE_MOCK"
                  />
                  <span
                    className="MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root"
                  >
                    20 min
                  </span>
                </div>
              </div>
            </div>
            <div
              className="textWithIcon card-header-footer"
            >
              <img
                alt="All meds covered"
                src="IMAGE_MOCK"
              />
              <span
                className="MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root"
              >
                All meds covered
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            className="MuiAccordion-region"
            role="region"
          >
            <div
              className="MuiAccordionDetails-root sc-bwzfXH hPCock css-15v22id-MuiAccordionDetails-root"
            >
              <div
                className="textWithIcon undefined"
              >
                <img
                  alt="Hyrdoxyzine hydrochloride"
                  src="IMAGE_MOCK"
                />
                <span
                  className="MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root"
                >
                  Hyrdoxyzine hydrochloride
                </span>
                <span
                  className="MuiTypography-root MuiTypography-caption textWithIcon-sideText css-1393mlh-MuiTypography-root"
                >
                  AED 150.00
                </span>
              </div>
              <div
                className="textWithIcon undefined"
              >
                <img
                  alt="Glimepiride-rosiglitazone"
                  src="IMAGE_MOCK"
                />
                <span
                  className="MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root"
                >
                  Glimepiride-rosiglitazone
                </span>
                <span
                  className="MuiTypography-root MuiTypography-caption textWithIcon-sideText css-1393mlh-MuiTypography-root"
                >
                  AED 150.00
                </span>
              </div>
              <div
                className="textWithIcon undefined"
              >
                <img
                  alt="Novolin 70/30"
                  src="IMAGE_MOCK"
                />
                <span
                  className="MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root"
                >
                  Novolin 70/30
                </span>
                <span
                  className="MuiTypography-root MuiTypography-caption textWithIcon-sideText css-1393mlh-MuiTypography-root"
                >
                  AED 150.00
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-fwdjlz-MuiDivider-root"
  />,
  <div
    className="styledComponents__Container-d2us26-1 cpOZRs"
  />,
  <div
    className="card-footer sc-htpNat rbDpa"
  >
    <div
      className="textWithIcon card-footer-title"
    >
      <span
        className="MuiTypography-root MuiTypography-caption css-1393mlh-MuiTypography-root"
      >
        Your est. co-payment*
      </span>
      <span
        className="MuiTypography-root MuiTypography-caption textWithIcon-sideText css-1393mlh-MuiTypography-root"
      >
        AED 50.00
      </span>
    </div>
    <span
      className="MuiTypography-root MuiTypography-caption card-footer-subtitle css-1393mlh-MuiTypography-root"
    >
      *Final amounts subject to insurance coverage
    </span>
  </div>,
]
`;
